### PR TITLE
Feature sessions

### LINKED ISSUE #12: [ui] : Create Sessions Screen & My Sessions functionality
- Create Sessions Screen as per the guideline - [Sessions Screen Design Adobe XD Link](https://xd.adobe.com/view/dd5d0245-b92b-4678-9d4a-48b3a6f48191-880e/screen/bfed7462-5ae7-4a23-8694-9e2d07487727/specs/)
- Implement "Show My Sessions" functionality - [My Sessions Design Adobe XD Link] (https://xd.adobe.com/view/dd5d0245-b92b-4678-9d4a-48b3a6f48191-880e/screen/ed33fe5e-dee5-435d-aac8-049fd85e764c/specs/)
- Follow contribution guidelines laid out in our [contribution guide](https://github.com/droidconKE/droidconKE2022ReactNative/blob/develop/CONTRIBUTING.md)

**Additional context**

- Sessions

Screenshot 2022-09-12 at 23 24 57

- My Session

Screenshot 2022-09-12 at 23 33 08

### PR DESCRIPTION
**Pull request by makunomark**
# Description
Adds a schedule of sessions screen with a date date selector to toggle dates. By default the selected date is the first date from the loaded schedule. Would be nice to have a feature that preselects the date in case today is one of the events dates. 

image

Work out of scope for this pr includes:
1. Filter functionality
2. API integration
3. Bookmark functionality

Partially Fixes #12 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
By running the app on iOS emulator

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/components/cards/SessionsVerticalList.tsx [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        <View style={styles.listHolder}>
+          {props.items.map((item, index) => (
+            <>
+              <SessionsVerticalListCard {...item} />
+              {index < props.items.length - 1 && (
+                <View style={styles.cardContainer}>
+                  <SessionListSeparator
+                    color={
+                      index % 2 !== 0 ? "rgba(255, 110, 77, 1)" : "#7de1c3"
+                    }
```
Comment (brianwachira) [inline src/components/cards/SessionsVerticalList.tsx]: Uhm I have noticed the colour` #7de1c3"` has been defined as `DROIDCONKE_LIGHT_GREEN` in the `Colors.ts` file. 
Could you also define the `rgba(255, 110, 77, 1)` inside the Colors.ts file?

File: src/components/cards/SessionsVerticalListCard.tsx [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+            <Text style={styles.durationAndVenue}>
+              {`${formatTime(props.start_date_time, true)} - ${formatTime(
+                props.end_date_time,
+                true
+              )} | ${props.rooms[0].title.toUpperCase()}`}
+            </Text>
+          )}
+          {props.speakers.length > 0 && (
+            <View style={styles.speakerContainer}>
+              <AndroidIcon width={17} color="rgba(0, 12, 235, 1)" />
```
Comment (brianwachira) [inline src/components/cards/SessionsVerticalListCard.tsx]: Uhm you can define this `rgba(0, 12, 235, 1)` colour separately on the `constants/Colors.ts` file instead?

File: src/components/cards/SessionsVerticalListCard.tsx [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+              <Text style={styles.speakerName}>{props.speakers[0]?.name}</Text>
+            </View>
+          )}
+        </View>
+      </View>
+      <Star color={"#000ceb"} />
```
Comment (brianwachira) [inline src/components/cards/SessionsVerticalListCard.tsx]: Uhm this `#000ceb` colour has already been defined on the `constants/Colors.ts`  as `DROIDCONKE_BLUE`. Can you use that 1 instead?

File: src/components/cards/SessionsVerticalListCard.tsx [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+  container: {
+    backgroundColor: "white",
+    borderRadius: 10,
+    flexDirection: "row",
+    padding: 16,
+  },
+  startTime: {
+    fontFamily: fonts.MONTSERRAT_MEDIUM,
+    fontSize: 18,
+    color: "rgba(32, 32, 30, 1)",
```
Comment (brianwachira) [inline src/components/cards/SessionsVerticalListCard.tsx]: Uhm you can define this `rgba(32, 32, 30, 1)` colour separately on the constants/Colors.ts file instead?

File: src/components/cards/SessionsVerticalListCard.tsx [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+  },
+  meridiem: {
+    fontFamily: fonts.MONTSERRAT_MEDIUM,
+    fontSize: 15,
+    textAlign: "right",
+    color: "rgba(32, 32, 30, 1)",
```
Comment (brianwachira) [inline src/components/cards/SessionsVerticalListCard.tsx]: Uhm you can define this `rgba(32, 32, 30, 1)` colour separately on the constants/Colors.ts file instead?

File: src/components/cards/SessionsVerticalListCard.tsx [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+  },
+  programTitle: {
+    fontFamily: fonts.MONTSERRAT_BOLD,
+    fontSize: 18,
+  },
+  sessionTitle: {
+    marginTop: 24,
+    fontFamily: fonts.MONTSERRAT_REGULAR,
+    fontSize: 16,
+    color: "rgba(32, 32, 30, 1)",
```
Comment (brianwachira) [inline src/components/cards/SessionsVerticalListCard.tsx]: Uhm you can define this `rgba(32, 32, 30, 1)` colour separately on the constants/Colors.ts file instead?

File: src/components/cards/SessionsVerticalListCard.tsx [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+  },
+  durationAndVenue: {
+    marginTop: 14,
+    fontFamily: fonts.MONTSERRAT_REGULAR,
+    color: "rgba(112, 112, 112, 1)",
```
Comment (brianwachira) [inline src/components/cards/SessionsVerticalListCard.tsx]: Uhm you can define this `rgba(112, 112, 112, 1)` colour separately on the constants/Colors.ts file instead?

File: src/components/cards/SessionsVerticalListCard.tsx [hunk 8]
```diff
@@ -0,0 +1,122 @@
...
+  },
+  speakerContainer: {
+    flexDirection: "row",
+    marginTop: 4,
+  },
+  speakerName: {
+    fontSize: 14,
+    marginLeft: 12,
+    color: "rgba(0, 12, 235, 1)",
```
Comment (brianwachira) [inline src/components/cards/SessionsVerticalListCard.tsx]: Uhm you can define this `rgba(0, 12, 235, 1)` colour separately on the constants/Colors.ts file instead?

File: src/components/dateToggle/DateToggle.tsx [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+    <View style={[styles.container, selected && styles.containerSelected]}>
+      <Text style={[styles.date, selected && styles.textSelected]}>{date}</Text>
+      <Text style={[styles.day, selected && styles.textSelected]}>{day}</Text>
+    </View>
+  );
+}
+
+const styles = StyleSheet.create({
+  container: {
+    backgroundColor: "rgba(125, 225, 195, 0.11)",
```
Comment (brianwachira) [inline src/components/dateToggle/DateToggle.tsx]: Uhm you can define this `rgba(125, 225, 195, 0.11)` colour separately on the constants/Colors.ts file instead?

File: src/components/dateToggle/DateToggle.tsx [hunk 10]
```diff
@@ -0,0 +1,58 @@
...
+    borderRadius: 5,
+    display: "flex",
+    justifyContent: "center",
+    padding: 12,
+  },
+  containerSelected: {
+    backgroundColor: "rgba(255, 110, 77, 1)",
```
Comment (brianwachira) [inline src/components/dateToggle/DateToggle.tsx]: Uhm you can define this `rgba(255, 110, 77, 1)` colour separately on the constants/Colors.ts file instead?

File: src/components/dateToggle/DateToggle.tsx [hunk 11]
```diff
@@ -0,0 +1,58 @@
...
+  },
+  date: {
+    fontSize: 18,
+    fontFamily: fonts.MONTSERRAT_BOLD,
+    color: "rgba(32, 32, 30, 1)",
```
Comment (brianwachira) [inline src/components/dateToggle/DateToggle.tsx]: Uhm you can define this `rgba(32, 32, 30, 1)` colour separately on the constants/Colors.ts file instead?

File: src/components/dateToggle/DateToggle.tsx [hunk 12]
```diff
@@ -0,0 +1,58 @@
...
+    alignSelf: "flex-start",
+  },
+  day: {
+    fontSize: 11,
+    color: "rgba(112, 112, 112, 1)",
```
Comment (brianwachira) [inline src/components/dateToggle/DateToggle.tsx]: Uhm you can define this `rgba(112, 112, 112, 1)` colour separately on the constants/Colors.ts file instead?

File: src/components/iconSwitch/Thumb.tsx [hunk 13]
```diff
@@ -0,0 +1,101 @@
...
+          style={[
+            styles.pressedIndicator,
+            {
+              backgroundColor: color,
+              left: pressedIndicatorPosition,
+              width: size * 2,
+              borderRadius: size,
+              top: 0 - radius,
+              aspectRatio: 1,
+            },
```
Comment (brianwachira) [inline src/components/iconSwitch/Thumb.tsx]: Uhm can these styles be defined separately?

File: src/components/iconSwitch/Thumb.tsx [hunk 14]
```diff
@@ -0,0 +1,101 @@
...
+      <Animated.View
+        style={[
+          styles.thumb,
+          {
+            backgroundColor: color,
+            left: position,
+            width: size,
+            borderRadius: radius,
+            aspectRatio: 1,
+          },
```
Comment (brianwachira) [inline src/components/iconSwitch/Thumb.tsx]: Uhm can these styles be defined separately?

File: src/screens/SessionsScreen.tsx [hunk 15]
```diff
@@ -1,15 +1,1470 @@
...
-export default SessionsScreen;
\ No newline at end of file
+const SessionsScreen = (props: SessionsScreenProps) => {
+  const { schedule } = useAppSelector((state) => state.schedule);
+  const [dates, setDates] =
+    useState<Pick<DateToggleListProps, "items"> | undefined>();
+  const [mySessions, setMySessions] = useState<boolean>(false);
+  const [sessions, setSessions] = useState<{ items: Session[] } | undefined>();
+  const [selectedDate, setSelectedDate] = useState<string | undefined>();
+  // const { selectedDay } = useAppSelector((state) => state.schedule.selectedDay);
```
Comment (brianwachira) [inline src/screens/SessionsScreen.tsx]: Uhm incase this is not going to be used in the future, you can remove it😅​

File: src/screens/SessionsScreen.tsx [hunk 16]
```diff
@@ -1,15 +1,1470 @@
...
+          <View style={{ paddingBottom: 20 }}>
+            <View style={styles.belowHeader}>
+              {dates && <DateToggleList {...dates} onChange={onDayChange} />}
+              <View style={styles.switchHolder}>
+                <SwitchWithIcons
+                  value={mySessions}
+                  onValueChange={onValueChange}
+                  trackColor={{ true: "#FF6E4D", false: "#20201E" }}
+                  thumbColor={{ true: "#FFFFFF", false: "#FFFFFF" }}
+                  iconColors={{ true: "#20201E", false: "#FF6E4D" }}
```
Comment (brianwachira) [inline src/screens/SessionsScreen.tsx]: So I have seen these colors have already been defined in `constants/Colors.ts` file
`#FF6E4D - DROIDCONKE_BRICK_RED`
`#20201E - DROIDCONKE_BLACK`
`#FFFFF - DROIDCONKE_WHITE`

You can replace the hexcodes with the appropriate defined color.

File: src/screens/SessionsScreen.tsx [hunk 17]
```diff
@@ -1,15 +1,1470 @@
...
+  function onDayChange(day: string) {
+    setSelectedDate(day);
+  }
+
+  return (
+    <View>
+      <SafeAreaView style={styles.paddingVertical}>
+        <MainHeader />
+        <ScrollView>
+          <View style={{ paddingBottom: 20 }}>
```
Comment (brianwachira) [inline src/screens/SessionsScreen.tsx]: Please define this paddingBottom style separately

File: src/screens/SessionsScreen.tsx [hunk 18]
```diff
@@ -1,15 +1,1470 @@
...
+};
+
+const styles = StyleSheet.create({
+  paddingVertical: {
+    paddingVertical: 20,
+  },
+  title: {
+    fontFamily: fonts.MONTSERRAT_BOLD,
+    fontSize: 18,
+    color: "rgba(0, 12, 235, 1)",
```
Comment (brianwachira) [inline src/screens/SessionsScreen.tsx]: Uhm could you define the `rgba(0, 12, 235, 1)` on `constants/colors.ts` file instead?

**Review by brianwachira — CHANGES_REQUESTED** [review]

**Review by brianwachira — CHANGES_REQUESTED** [review]
Good work man. 
Just a few changes ( mostly defining colors in `constants/colors.ts` ) that is needed.

File: src/components/iconSwitch/Thumb.tsx (hunk 14 above)
Comment (makunomark) [reply src/components/iconSwitch/Thumb.tsx]: color, position, size and radius are calculated. May be hard to move them to styles.

File: src/components/iconSwitch/Thumb.tsx (hunk 13 above)
Comment (makunomark) [reply src/components/iconSwitch/Thumb.tsx]: color, pressedIndicatorPosition, size, and radius are either calculated or passed as props. May be hard to move them.

**Review by brianwachira — APPROVED** [review]

**Pull request merged**